app/scripts/views/settings/communication_preferences.js

```diff
@@ -39,19 +40,23 @@ function (Cocktail, Xss, Constants, MarketingEmailErrors, BaseView, FormView,
     beforeRender: function () {
       var self = this;
       var emailPrefs = self.getMarketingEmailPrefs();
-
       return emailPrefs.fetch()
         .fail(function (err) {
           if (MarketingEmailErrors.is(err, 'UNKNOWN_EMAIL')) {
             // user has not yet opted in to Basket yet. Ignore.
             return;
           }
-
-          // error fetching the email preferences, show the
-          // error and hide the spinner.
-          self.$('.spinner').hide();
-          self.displayError(err);
-          return false;
+          if (MarketingEmailErrors.is(err, 'UNKNOWN_ERROR')) {
+            self._error = self.translateError(MarketingEmailErrors.toError('SERVICE_UNAVAILABLE'));
+          } else {
+            self._error = self.translateError(err);
+          }
+          err = self._normalizeError(err);
+          var errorString = Metrics.prototype.errorToId(err);
+          // Add status code to metrics data, to differentiate between
+          // 400 and 500
+          errorString = errorString + '.' + err.code;
```

Comment:
Even though we think there is an err.code, we can't really depend on it, so let's change `err.code` to `(err.code || 'unknown')`